### PR TITLE
Subscriptions: update "verify your email" wall copy

### PR DESCRIPTION
**Pull request by simison**
Updates copy to be more about "confirming their subscription" rather than "confirming email" per-se, and aligns copy with email they received.

See https://github.com/Automattic/wp-calypso/issues/84773#issuecomment-1845026885

Doesn't update the actual modal where we also nudge to check email.

## Proposed changes:

* Update copy
* Add email to the copy

Before
Screenshot 2023-12-19 at 14 47 38

After
Screenshot 2023-12-19 at 14 46 10

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Subscribe to post requiring subscription, but don't confirm your email. 
* Email does not get cached for others to see because the whole page isn't cached (https://github.com/Automattic/jetpack/pull/34242#issuecomment-1838479034)

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php [hunk 1]
```diff
@@ -1114,9 +1114,18 @@ function is_user_auth() {
...
+	if ( is_wpcom() || false !== apply_filters( 'jetpack_auto_fill_logged_in_user', false ) ) {
+		$current_user    = wp_get_current_user();
+		$subscribe_email = ! empty( $current_user->user_email ) ? $current_user->user_email : '';
+	}
+
+	$access_heading = esc_html__( 'Confirm your subscription to continue reading', 'jetpack' );
 
-	$subscribe_text = esc_html__( 'Please check your inbox to confirm your subscription.', 'jetpack' );
+	/* translators: %s: email address */
+	$subscribe_text = wp_kses( sprintf( __( 'Head to your inbox and confirm your email address %s.', 'jetpack' ), $subscribe_email ), array() );
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php]: Do we need `wp_kses` here? Could we use `esc_html` here instead? 

```suggestion
	$subscribe_text = sprintf( esc_html__( 'Head to your inbox and confirm your email address %s.', 'jetpack' ), $subscribe_email );
```

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php [hunk 2]
```diff
@@ -1114,9 +1114,18 @@ function is_user_auth() {
  * @return string
  */
 function get_paywall_blocks_subscribe_pending() {
-	$access_heading = esc_html__( 'Verify your email to continue reading', 'jetpack' );
+	$subscribe_email = Jetpack_Memberships::get_current_user_email();
+
+	/** This filter is documented in modules/contact-form/grunion-contact-form.php */
+	if ( is_wpcom() || false !== apply_filters( 'jetpack_auto_fill_logged_in_user', false ) ) {
```
Comment (lezama) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:1120]: I think we should use the `Host` class for these checks, but we can do it in a separate PR

**Review by lezama — APPROVED** [review]
